Panic and failure when parsing K8 HPA resource metric blocks

**Describe the bug**
When running tfsec against my module it fails with a panic null reference error.

I was able to narrow it down using the --debug command to a single file, one handling horizontal pod autoscaler resources for Kubernetes. Then I removed blocks from those HPA resource until the error went away, it turns out it's the metric block that seems to be causing it: 

```hcl
metric {
      type = "Resource"
      resource {
        name = "memory"
        target {
          type                = "Utilization"
          average_utilization = "80"
        }
      }
    }
```

Both of the metric blocks I have needed to be removed to get the panic to go away. (See tf code below for full HPA resource)

**To Reproduce**
Have tfsec parse a file with a HPA with a metric block like what was listed above. This occurs in the latest release 1.13. In the previous I was using it did not occur, 0.63.1.

**Expected behavior**
A full clean TFSec run

**Screenshots/Output**
Here is the panic stack trace:
```panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10347867c]

goroutine 1 [running]:
github.com/aquasecurity/tfsec/internal/pkg/block.extendReference(...)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/reference.go:22
github.com/aquasecurity/tfsec/internal/pkg/block.NewAttribute(0x1400040c5a0, 0x1400017f3d0, {0x1035c8fee, 0x4}, {{0x103881698, 0x14000400ee0}, {0x103878380, 0x0}, 0x1, 0x0, ...}, ...)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/attribute.go:29 +0x6c
github.com/aquasecurity/tfsec/internal/pkg/block.New(0x1400040e000, 0x1400017f3d0, 0x0, 0x1400040c280)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/block.go:99 +0xa3c
github.com/aquasecurity/tfsec/internal/pkg/block.New(0x14000403ee0, 0x1400017f3d0, 0x0, 0x14000407f40)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/block.go:85 +0xe4c
github.com/aquasecurity/tfsec/internal/pkg/block.New(0x14000403c70, 0x1400017f3d0, 0x0, 0x14000407cc0)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/block.go:85 +0xe4c
github.com/aquasecurity/tfsec/internal/pkg/block.New(0x14000403450, 0x0, 0x0, 0x0)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/block.go:85 +0xe4c
github.com/aquasecurity/tfsec/internal/pkg/parser.(*Parser).parseDirectoryFiles(0x140003647e0, {0x140004d0900, 0x9, 0x10})
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/parser/parser.go:59 +0x3b0
github.com/aquasecurity/tfsec/internal/pkg/parser.(*Parser).ParseDirectory(0x140003647e0)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/parser/parser.go:93 +0x6bc
github.com/aquasecurity/tfsec/internal/app/tfsec/cmd.glob..func1(0x103dcb440, {0x103e36af0, 0x0, 0x0})
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/app/tfsec/cmd/root.go:108 +0x8a0
github.com/spf13/cobra.(*Command).execute(0x103dcb440, {0x140000961e0, 0x0, 0x0})
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/vendor/github.com/spf13/cobra/command.go:856 +0x668
github.com/spf13/cobra.(*Command).ExecuteC(0x103dcb440)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/vendor/github.com/spf13/cobra/command.go:974 +0x410
github.com/spf13/cobra.(*Command).Execute(...)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/vendor/github.com/spf13/cobra/command.go:902
main.main()
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/cmd/tfsec/main.go:11 +0x30
```
And the debug output:
```
[DEBUG][2022-02-10 12:14:26.144007 -0500 EST m=+0.002819335] Loading custom checks...
[DEBUG][2022-02-10 12:14:26.144091 -0500 EST m=+0.002903543] Using the default custom check folder
[DEBUG][2022-02-10 12:14:26.144094 -0500 EST m=+0.002906251] custom check directory set to /Users/david/Code/Dozuki/CloudPrem-Infra/cloudprem/logical/.tfsec
[DEBUG][2022-02-10 12:14:26.144103 -0500 EST m=+0.002915126] Custom checks loaded
[DEBUG][2022-02-10 12:14:26.144105 -0500 EST m=+0.002917626] checking for tfvars files using glob: /Users/david/Code/Dozuki/CloudPrem-Infra/cloudprem/logical/*.tfvars
[DEBUG][2022-02-10 12:14:26.144188 -0500 EST m=+0.003000751] Starting parser...
[DEBUG][2022-02-10 12:14:26.144194 -0500 EST m=+0.003005918] Finding Terraform subdirectories...
[DEBUG][2022-02-10 12:14:26.144311 -0500 EST m=+0.003122918] Found qualifying subdirectory containing .tf files: /Users/david/Code/Dozuki/CloudPrem-Infra/cloudprem/logical
[DEBUG][2022-02-10 12:14:26.144315 -0500 EST m=+0.003126960] Beginning parse for directory '/Users/david/Code/Dozuki/CloudPrem-Infra/cloudprem/logical'...
[DEBUG][2022-02-10 12:14:26.149339 -0500 EST m=+0.008151293] Added 5 blocks from /Users/david/Code/Dozuki/CloudPrem-Infra/cloudprem/logical/replicated.tf...
[DEBUG][2022-02-10 12:14:26.149434 -0500 EST m=+0.008246501] Added 4 blocks from /Users/david/Code/Dozuki/CloudPrem-Infra/cloudprem/logical/scaling.tf...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10114867c]

goroutine 1 [running]:
github.com/aquasecurity/tfsec/internal/pkg/block.extendReference(...)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/reference.go:22
github.com/aquasecurity/tfsec/internal/pkg/block.NewAttribute(0x140002cd540, 0x140001f4650, {0x101298fee, 0x4}, {{0x101551698, 0x14000624aa0}, {0x101548380, 0x0}, 0x1, 0x0, ...}, ...)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/attribute.go:29 +0x6c
github.com/aquasecurity/tfsec/internal/pkg/block.New(0x1400047a4e0, 0x140001f4650, 0x0, 0x140002ccf00)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/block.go:99 +0xa3c
github.com/aquasecurity/tfsec/internal/pkg/block.New(0x1400047a410, 0x140001f4650, 0x0, 0x140002cc8c0)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/block.go:85 +0xe4c
github.com/aquasecurity/tfsec/internal/pkg/block.New(0x1400047a1a0, 0x140001f4650, 0x0, 0x140002cc3c0)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/block.go:85 +0xe4c
github.com/aquasecurity/tfsec/internal/pkg/block.New(0x14000429930, 0x0, 0x0, 0x0)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/block/block.go:85 +0xe4c
github.com/aquasecurity/tfsec/internal/pkg/parser.(*Parser).parseDirectoryFiles(0x140003da7e0, {0x1400007c900, 0x9, 0x10})
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/parser/parser.go:59 +0x3b0
github.com/aquasecurity/tfsec/internal/pkg/parser.(*Parser).ParseDirectory(0x140003da7e0)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/pkg/parser/parser.go:93 +0x6bc
github.com/aquasecurity/tfsec/internal/app/tfsec/cmd.glob..func1(0x101a9b440, {0x140003f35e0, 0x0, 0x1})
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/internal/app/tfsec/cmd/root.go:108 +0x8a0
github.com/spf13/cobra.(*Command).execute(0x101a9b440, {0x14000134010, 0x1, 0x1})
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/vendor/github.com/spf13/cobra/command.go:856 +0x668
github.com/spf13/cobra.(*Command).ExecuteC(0x101a9b440)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/vendor/github.com/spf13/cobra/command.go:974 +0x410
github.com/spf13/cobra.(*Command).Execute(...)
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/vendor/github.com/spf13/cobra/command.go:902
main.main()
	/private/tmp/tfsec-20220209-39631-1a8frw6/tfsec-1.1.3/cmd/tfsec/main.go:11 +0x30
```

**System Info**
<!-- complete the following -->
 - tfsec version: `1.13`
 - terraform version: `1.0.11`
 - OS: `MacOS Monterey` but also occurring when running view GitHub actions on an ubuntu host.

**Example Code**
This is the entire HPA resource:
```hcl
resource "kubernetes_horizontal_pod_autoscaler" "app" {
  depends_on = [helm_release.replicated]
  metadata {
    name = "app-hpa"
    # Terraform magic required to find the randomly created replicated namespace so we can install the HPAs in the right place.
    namespace = coalesce([for i, v in data.kubernetes_all_namespaces.allns.namespaces : try(regexall("replicated\\-.*", v)[0], "")]...)
  }

  spec {
    min_replicas = 2
    max_replicas = 10

    scale_target_ref {
      kind        = "Deployment"
      name        = "app-deployment"
      api_version = "apps/v1"
    }

    metric {
      type = "Resource"
      resource {
        name = "memory"
        target {
          type                = "Utilization"
          average_utilization = "80"
        }
      }
    }

    metric {
      type = "Resource"
      resource {
        name = "cpu"
        target {
          type                = "Utilization"
          average_utilization = "50"
        }
      }
    }
  }
}
```

--- a/internal/pkg/block/block.go
+++ b/internal/pkg/block/block.go
@@ -51,7 +51,10 @@ func New(hclBlock *hcl.Block, ctx *Context, moduleBlock *Block, parentBlock *Blo
 	)
 
 	var parts []string
-	if hclBlock.Type != "resource" {
+	// if there are no labels then use the block type
+	// this is for the case where "special" keywords like "resource" are used
+	// as normal block names in top level blocks - see issue #1528 for an example
+	if hclBlock.Type != "resource" || len(hclBlock.Labels) == 0 {
 		parts = append(parts, hclBlock.Type)
 	}
 	parts = append(parts, hclBlock.Labels...)
@@ -78,6 +81,18 @@ func New(hclBlock *hcl.Block, ctx *Context, moduleBlock *Block, parentBlock *Blo
 		metadata:    metadata,
 	}
 
+	children := getChildren(hclBlock, ctx, moduleBlock, b)
+
+	b.childBlocks = children
+
+	for _, attr := range b.createAttributes() {
+		b.attributes = append(b.attributes, NewAttribute(attr, ctx, moduleName, metadata, ref))
+	}
+
+	return &b
+}
+
+func getChildren(hclBlock *hcl.Block, ctx *Context, moduleBlock *Block, b Block) Blocks {
 	var children Blocks
 	switch body := hclBlock.Body.(type) {
 	case *hclsyntax.Body:
@@ -92,14 +107,7 @@ func New(hclBlock *hcl.Block, ctx *Context, moduleBlock *Block, parentBlock *Blo
 			}
 		}
 	}
-
-	b.childBlocks = children
-
-	for _, attr := range b.createAttributes() {
-		b.attributes = append(b.attributes, NewAttribute(attr, ctx, moduleName, metadata, ref))
-	}
-
-	return &b
+	return children
 }
 
 func (b *Block) ID() string {

